### PR TITLE
[RFC] Build against Windows 10 SDK

### PR DESCRIPTION
**Pull request by parkerlamb**
Xenia currently builds on Windows 10, though premake defaults to using the Windows 8.1 SDK when generating VS projects. This PR allows for building Xenia with only a Windows 10 SDK installed.

There are two issues which need to be resolved before merging:

1. The code in this PR selects the most recent Windows 10 SDK on the system by looking at the registry. This is flexible but tends to imply that Xenia supports _all_ of the SDKs which may not be a good idea. Alternatively, we could pick a version and stick with it, though at the rate Microsoft puts out new Windows 10 builds it'd probably have to get bumped pretty frequently.

2. The premake version needs to be bumped. The binary at /tools/build/bin/premake5.exe is fairly out of date and doesn't report OS versions correctly (it reports Windows 10 as Windows 8 on my system). It could be replaced with a newer version or it could be deleted and the project could build premake from third_party/premake_core automatically. The premake build system needs some tweaks to work on Windows 10 (pretty much identical to what's been done here) but since Xenia has its own version of premake anyway this doesn't seem unreasonable. Building premake from source will probably be required for Linux builds so it may make sense to do it now.

Other things: 
- XInputEnable() has been deprecated under Windows 10, though from looking through the docs it doesn't seem like it was being used correctly here anyway. XInput still works fine with XInputEnable() being removed.
- flags({"WinMain"}) is no longer required for newer versions of premake, kind("WindowedApp") now implies this.

Comments welcome!

### CODE REVIEW

**Comment by parkerlamb** [comment]
Forgot to mention, this fixes #693.

**Comment by DrChat** [comment]
It appears that your change from "os.is" to "os.istarget" has broken the latest builds.
Have you tested this against the premake version we're using?

**Comment by parkerlamb** [comment]
I tested this against the latest commit (52b3b5f) in xenia-project/premake-core. This is not the commit Xenia is currently tracking for the submodule, which is 61d799c8.

The 61d799c8 version does not report OS versions correctly, which is why it needs to be updated. "os.is" has been deprecated in the 52b3b5f version, hence the errors.

The buildbot will fail because the build script prioritizes the precompiled binary at /tools/build/bin/premake5.exe, which is the older version.

As I mentioned above, a decision needs to be made as to whether the precompiled binary should be updated or just move to building premake from source. The submodule commit needs to be bumped either way.

The lint errors are my fault, I'll fix them once everything else is sorted out.

**Comment by DrChat** [comment]
Ah - don't build against premake-core's master branch. We're building against the release branch to keep bleeding-edge bugs away.

**Comment by parkerlamb** [comment]
This PR needs the premake-core master branch to work because of the OS version reporting issue (at least up to commit 0a7b45ce which has the patch for it). If you don't want Xenia to track a newer commit then this PR will have to wait until the premake devs release a new version.

**Comment by Margen67** [comment]
Why was this closed?

**Comment by parkerlamb** [comment]
It's basically a Visual Studio issue. There's a somewhat philosophical discussion on the Premake bug tracker here: [https://github.com/premake/premake-core/issues/935](https://github.com/premake/premake-core/issues/935).

TLDR:
Visual Studio defaults to using the Windows 8.1 SDK when one isn't specified in the config files. There isn't really a good way to determine which SDK to use for Windows 10 because users may have different ones installed on their systems, plus Microsoft pushes a new Windows build every 6 months which means the recent version constantly changes. If we picked a particular SDK to target, it would need to be constantly bumped to match or be out of date on newer Windows builds.

It's my view that this is something Microsoft needs to fix, not hack-fixed in the project files. If you (or anyone else) wants to continue with this PR, feel free to use my changes.
